app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,17 @@
+# Join table to keep track of favorite books. Belong to user and to book.
+class FavoritesController < ApplicationController
+  def create
+    authorize :favorite
+    book = Book.find(params[:book_id])
+    favorite = book.favorites.build
+    favorite.user_id = current_user.id
```

Comment:
create in User has_many through relationship to query books directly like, and the do this, current_user.favorite_books << book